src/Core/Parser/SyntaxTree/Expression/CastingExpressionNode.php

```diff
@@ -113,8 +115,16 @@ protected static function convertToArray($variable)
         if (is_array($variable)) {
             return $variable;
         }
-        if (is_string($variable) && strpos($variable, ',')) {
-            return array_map('trim', explode(',', $variable));
+        if (is_string($variable)) {
+            if (preg_match('/^[\[\{].*[\]\}]$/', $variable)) {
```

Comment:
I fear this check must be dropped since simple literals like `'null'`, `1` are also valid JSON, not only lists and objects/maps.

Reply:
Geh, the point here is conversion to array, so it's perfectly fine like this. :facepalm: